types/src/share.rs

```diff
@@ -57,6 +57,32 @@ impl Share {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(transparent)]
+struct RawNamespacedShares {
+    rows: Option<Vec<NamespacedRow>>,
+}
+
+impl From<RawNamespacedShares> for NamespacedShares {
+    fn from(value: RawNamespacedShares) -> Self {
+        Self {
+            rows: value.rows.unwrap_or(vec![]),
+        }
+    }
+}
+
+impl From<NamespacedShares> for RawNamespacedShares {
+    fn from(value: NamespacedShares) -> Self {
+        let rows = if !value.rows.is_empty() {
+            Some(value.rows)
+        } else {
+            None
+        };
```

Comment:
in this case we can avoid `not`

```suggestion
        let rows = if value.rows.is_empty() {
            None
        } else {
            Some(value.rows)
        };
```